### ISSUE TITLE
Rule proposal: `no-duplicate-assign`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#1304]
## Fail

````js
a = a = 1;
a = b = a = 1;
let a = b = b = 1;
````

## Pass

```js
a = 1;
a = b = 1;
let a = b = 1;
```

Happend to me recently as part of a semi-automated refactoring.

**Comment by fisker** [comment]
[`no-multi-assign`](https://eslint.org/docs/rules/no-multi-assign)?

**Comment by silverwind** [comment]
Not suitable, that rule completely forbids multiple assignment which I stil want to allow, just not usless duplicate ones.

**Comment by sindresorhus** [comment]
I'm -0 on this. I can see how the rule could be useful when you use multi-assign, but I also strongly believe multi-assign is an anti-pattern.

**Comment by sindresorhus** [comment]
I'm going to pass on this. I suggest you try to open an issue on ESLint. Maybe they're willing to add it.
